crates/rustic_core/src/commands/forget.rs

```diff
@@ -2,33 +2,43 @@
 
 use chrono::{DateTime, Datelike, Duration, Local, Timelike};
 use derivative::Derivative;
+use derive_setters::Setters;
 use serde::Deserialize;
 use serde_with::{serde_as, DisplayFromStr};
 
 use crate::{
-    repository::Open, Id, ProgressBars, Repository, RusticResult, SnapshotFile, SnapshotGroup,
-    SnapshotGroupCriterion, StringList,
+    repofile::SnapshotFile, repository::Open, Id, ProgressBars, Repository, RusticResult,
+    SnapshotGroup, SnapshotGroupCriterion, StringList,
 };
 
 type CheckFunction = fn(&SnapshotFile, &SnapshotFile) -> bool;
 
 #[derive(Debug)]
+/// A newtype for `[Vec<ForgetGroup>]`
 pub struct ForgetGroups(pub Vec<ForgetGroup>);
 
 #[derive(Debug)]
+/// All snapshots of a group with group and forget information
 pub struct ForgetGroup {
+    /// The group
     pub group: SnapshotGroup,
```

Comment:
`ForgetGroups` is just a newtype for `Vec<ForgetGroup>`and is the result you get when you call `Repository.get_forget_groups()` which is part of the Repository API. Therefore I made all these structs public. Also, `SnapshotGroup` is already used in `src/commands/forget`. You are right that so far only methods of `SnapshotGroup` are used in the CLI. But this may change when a GUI is used or when  we add #626, where a serialized output of `SnapshotGroup` will be returned. And I used the paradigm that all structs which are `Serialize` or `Deserialize` (and no fields are skipped for (de-)serialization) should only contain `pub` items - it just doesn't make sense to let users read/write it in the serialized form but forbid to read/write it directly...